src/watcher.js

```diff
@@ -42,9 +49,14 @@ export default function onSourceChange(
     proxyFileChanges({artifactsDir, onChange, filePath, shouldWatchFile});
   });
 
-  log.debug(`Watching for file changes in ${sourceDir}`);
+  log.debug(`Watching for file changes in ${watchFile || sourceDir}`);
+
   watcher.watch([], [sourceDir], Date.now());
 
+  if (watchFile) {
```

Comment:
To avoid reloading too often, when `watchFile` is specified, `sourceDir` should not be watched any more. Could you update the implementation and tests with this expectation?